### PR TITLE
Add Logger Tests

### PR DESCRIPTION
**Pull request by colin99d**
# Description

Bringing coverage for logging functionality from 0% to 98%.

# How has this been tested?

Ran tests

# Checklist:

- [x] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: tmp/MOCK_COMMIT_HASH_MOCK_SESSION_ID.2022-04-17_21.log [hunk 1]
```diff
@@ -0,0 +1,602 @@
+X|MOCK_COMMIT_HASH|MOCK_COMMIT_HASH|MOCK_COMMIT_HASH|MOCK_SESSION_ID|NA|2022-04-17T21:49:52-0400|openbb_terminal.parent_classes|menu|338|The command 'help' doesn't exist on the /alternative/ menu. - Traceback(most recent call last): - File openbb_terminal/parent_classes.py, line 334, in menu - self.queue = self.switch(an_input) - File cut/file/path/mock.py, line 1081, in __call__ - return self._mock_call(*args, **kwargs) - File cut/file/path/mock.py, line 1085, in _mock_call - return self._execute_mock_call(*args, **kwargs) - File cut/file/path/mock.py, line 1146, in _execute_mock_call - result = effect(*args, **kwargs) - File tests/openbb_terminal/alternative/test_alt_controller.py, line 102, in __call__ - raise SystemExit() - SystemExit
```
Comment (Chavithra) [inline tmp/MOCK_COMMIT_HASH_MOCK_SESSION_ID.2022-04-17_21.log:1]: I think we should make sure the test is not generating files.

**Review by Chavithra — CHANGES_REQUESTED** [review]
Looks like this is generating files.

File: tmp/MOCK_COMMIT_HASH_MOCK_SESSION_ID.2022-04-17_21.log (hunk 1 above)
Comment (colin99d) [reply tmp/MOCK_COMMIT_HASH_MOCK_SESSION_ID.2022-04-17_21.log]: Good catch. Fixed!

**Review by Chavithra — CHANGES_REQUESTED** [review]
Some issues I have found :
1. When I run this I still have files generated and not cleaned, i.e. : `dontreadme.txt` and `readme.txt`
2. A fixture called `remove_test_dir` is deploy globally and autoused ? why deploy globally a tool which is meant to solve one specific issue ?

My recommendations : 
- replace the fixture `remove_test_dir` with `tmp_path`, which is provided by default with `pytest`.
- use this `tmp_path` as `log_dir`, so the logs can be generated there during testing and it will be automatically cleaned by `pytest`

**Comment by colin99d** [comment]
Chavi's TODOs:

- Replace delete with temporary folder
- Refactor tests to not use the mock classes

File: tests/openbb_terminal/core/log/collection/test_log_sender.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    Settings,
+    AppSettings,
+    AWSSettings,
+    LogSettings,
+)
+
+settings = Settings(
+    app_settings=AppSettings(
+        commit_hash="MOCK_COMMIT_HASH",
+        name="MOCK_COMMIT_HASH",
```
Comment (Chavithra) [inline tests/openbb_terminal/core/log/collection/test_log_sender.py:13]: Not mocking `commit_hash`

File: tests/openbb_terminal/core/log/collection/test_log_sender.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        identifier="MOCK_COMMIT_HASH",
```
Comment (Chavithra) [inline tests/openbb_terminal/core/log/collection/test_log_sender.py:14]: Not mocking `commit_hash`

Comment (Chavithra) [inline tests/openbb_terminal/core/log/collection/test_log_sender.py:14]: Not mocking `commit_hash`

File: tests/openbb_terminal/core/log/collection/test_log_sender.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        session_id="MOCK_SESSION_ID",
+    ),
+    aws_settings=AWSSettings(
+        aws_access_key_id="MOCK_AWS_ACCESS_KEY_ID",
+        aws_secret_access_key="MOCK_AWS",  # pragma: allowlist secret
+    ),
+    log_settings=LogSettings(
+        directory=Path("."),
```
Comment (Chavithra) [inline tests/openbb_terminal/core/log/collection/test_log_sender.py:22]: On my setup it is still generating and leaving some file.
Probably because of this Path.
Let's mock the path so it won't generate any actual file.

Comment (Chavithra) [inline tests/openbb_terminal/core/log/collection/test_log_sender.py:22]: Let's mock the path so it won't generate any actual file.
On my setup it is still generating and leaving some file.

File: tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import os
+import pytest
+from openbb_terminal.core.log.generation import formatter_with_exceptions as fwe
+from openbb_terminal.core.log.generation.settings import AppSettings
+from openbb_terminal.core.config.constants import REPO_DIR
+
+app_settings = AppSettings(
+    commit_hash="MOCK_COMMIT_HASH",
+    name="MOCK_COMMIT_HASH",
```
Comment (Chavithra) [inline tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py:9]: `MOCK_COMMIT_HASH` ?

Comment (Chavithra) [inline tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py:9]: `MOCK_COMMIT_HASH` ?

File: tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    identifier="MOCK_COMMIT_HASH",
```
Comment (Chavithra) [inline tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py:10]: `MOCK_COMMIT_HASH` ?

Comment (Chavithra) [inline tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py:10]: `MOCK_COMMIT_HASH` ?

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -38,7 +38,7 @@ repos:
         entry: codespell
         args:
           [
-            "--ignore-words-list=zlot,ba,buil,coo,ether,hist,hsi,mape,navagation,operatio,pres,ser,yeld,shold,ist,varian,datas,AKE",
+            "--ignore-words-list=zlot,ba,buil,coo,ether,hist,hsi,mape,navagation,operatio,pres,ser,yeld,shold,ist,varian,datas,AKEi,fwe",
```
Comment (Chavithra) [inline .pre-commit-config.yaml:41]: Is this necessary for this PR ?

Comment (Chavithra) [inline .pre-commit-config.yaml:41]: Is this necessary for this PR ?

**Review by Chavithra — CHANGES_REQUESTED** [review]

File: tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    value = formatter.detect_terminal_message("The command doesn't exist on the")
+    assert value or os.sep != "/"
+    new_val = formatter.detect_terminal_message("boring message")
+    assert new_val is False
+
+
+def test_filter_log_line():
+    value = formatter.filter_log_line("stocks")
+    assert value
+    value = formatter.filter_log_line(
```
Comment (Chavithra) [inline tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py:57]: Here the object is not necessary since `filter_log_line` is a static method.
Same for the other static methods from `FormatterWithExceptions`.

File: .pre-commit-config.yaml (hunk 7 above)
Comment (Chavithra) [inline .pre-commit-config.yaml:41]: Is this change necessary for this PR ?

**Review by Chavithra — CHANGES_REQUESTED** [review]

File: tests/openbb_terminal/core/log/collection/test_log_sender.py (hunk 3 above)
Comment (colin99d) [reply tests/openbb_terminal/core/log/collection/test_log_sender.py:14]: I copied this from what you sent me. Let me know how I should change!

File: tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py (hunk 6 above)
Comment (colin99d) [reply tests/openbb_terminal/core/log/generation/test_formatter_with_exceptions.py:10]: I copied this from what you sent me. Let me know how I should change!

**Comment by DidierRLopes** [comment]
@Chavithra did you have time to look at this?

**Review by Chavithra — APPROVED** [review]
